File: Jumping on the Clouds/jumping-on-the-clouds.js
/**
 * Determine the minimum number of jumps it to jump from her starting position to the last cloud
 *
 * @param {array} c
 * @return {int}
 */
const jumpingOnClouds = (c) => {
  let jumps = 0;
  c = c.slice(1);
  while (c.length > 1) {
    let part = c[1] === 0 ? 2 : 1;
    c = c.slice(part);
    jumps++;
  }

  return jumps + c.length;
};

console.debug(jumpingOnClouds([0, 0, 0, 1, 0, 0])); // 3
console.debug(jumpingOnClouds([0, 0, 1, 0, 0, 1, 0])); // 4

File: Jumping on the Clouds/jumping-on-the-clouds.go
package main

import (
	"fmt"
)

func jumpingOnClouds(c []int32) int32 {
	jumps := 0
	c = c[1:]
	for len(c) > 1 {
		part := 1
		if c[1] == 0 {
			part = 2
		}
		c = c[part:]
		jumps++
	}

	return int32(jumps + len(c))
}

func main() {
	test1 := []int32{0, 0, 0, 1, 0, 0}
	test2 := []int32{0, 0, 1, 0, 0, 1, 0}

	fmt.Printf("jumpingOnClouds(%v) = %d\n", test1, jumpingOnClouds(test1)) // 3
	fmt.Printf("jumpingOnClouds(%v) = %d\n", test2, jumpingOnClouds(test2)) // 4
}